### ISSUE TITLE
Increase measurement error for Joystick's corner values

### ISSUE THREAD

**Issue by redisotschek** [MarcoFazioRandom/Virtual-Joystick-Godot#43]
If user wants to go straight sideways or up/down the joystick still works only in diagonal, because this check
```
if output.x < -0:
    Input.action_press(action_left, -output.x)
elif Input.is_action_pressed(action_left):
    Input.action_release(action_left)
```

returns true even for the smallest values and it's impossible to hit straight on 0

How about increasing value in check to 0.1 / -0.1?

This works perfectly
```
func _update_input_actions():
	if output.x < -0.1:
		Input.action_press(action_left, -output.x)
	elif Input.is_action_pressed(action_left):
		Input.action_release(action_left)
	if output.x > 0.1:
		Input.action_press(action_right, output.x)
	elif Input.is_action_pressed(action_right):
		Input.action_release(action_right)
	if output.y < -0.1:
		Input.action_press(action_up, -output.y)
	elif Input.is_action_pressed(action_up):
		Input.action_release(action_up)
	if output.y > 0.1:
		Input.action_press(action_down, output.y)
	elif Input.is_action_pressed(action_down):
		Input.action_release(action_down)
```

**Comment by MarcoFazioRandom** [comment]
Hello redisotschek! 

If I understand correctly, this behavior is not related to this project, as a real joystick would present the same problem (you can't move the joystick in a perfect single direction). 

The issue should be solved afterwards, after you take the value from the input action.

**Comment by MarcoFazioRandom** [comment]
I'm reopening this issue, after another user opened  #45 for the same problem, because I will commit a fix soon.

In short, I will use the value from `InputMap.action_get_deadzone()` instead of 0.

**Comment by MarcoFazioRandom** [comment]
Fixed with #46

**Comment by KuroyukiDev** [comment]
> Hello redisotschek!
> 
> If I understand correctly, this behavior is not related to this project, as a real joystick would present the same problem (you can't move the joystick in a perfect single direction).
> 
> The issue should be solved afterwards, after you take the value from the input action.

Actually redisoschek is correct on this @MarcoFazioRandom. From my own experience with trying to use your joystick for moving my game's character with a 4-directional movement animation, the Joystick is forcing the output variable's Vector2 value to completely disregard horizontal movement. 

Here's what I mean:
(Using normal Up/Down/Left/Right arrow keys)
```swift
const SPEED = 400

# physics_process()

# Get the input as a movement direction Vector2 and normalize it
var input_vector = Input.get_vector("ui_left", "ui_right", "ui_up", "ui_down")
input_vector = input_vector.normalized()

# Apply the movement direction and the speed to the velocity
velocity = input_vector * SPEED

# Set the animation based on the direction the object is moving 
# (Based on an 8 direction movement input which is what a joystick input method does)

var direction = ""

if velocity.y < 0:    
    direction = "up"
elif velocity.y > 0:
    direction = "down"

if velocity.x < 0:
    direction += "_left"
elif velocity.x > 0:
    direction += "_right"

# Apply the velocity to the object
move_and_slide()
```

However, your joystick is ignoring the use of the "up_left", "up_right" and "down_left", "down_right" directions when setting the `output` variable.... either that or it is doing something weird with the calculation that isn't returning the same values as we would expect from the Input.get_vector() function meaning it is pretty much useless to up for the movement of anything other than an object that doesn't use the movement direction to set its animations. As the joystick plugin is now the `output` variable would only be usable for setting the Up/Down animations of a character's animations while not supporting Left/Right animations if the joystick is dragged to the diagonal directions and the `output` variable doesn't even account for absolute Left/Right movement values (likely due to the strong bias towards the Up/Down directions in the calculation of the `output` variable).

If there is a variable you've added that returns the actual movement direction Vector2 value, I haven't seen it in your code anywhere. I've noticed several issues about this topic that were closed without any solution so there is a demand for a solution. Please add a solution for this for your users.
